xarray/core/concat.py

```diff
@@ -404,20 +406,51 @@ def process_subset_opt(opt, subset):
     return concat_over, equals, concat_dim_lengths
 
 
+# search for dataset containing all wanted variables
+# if no matching dataset is found return order from first dataset
+# and append missing variables at the end
+def _get_var_order(
```

Comment:
IIRC there's only one groupby test that explicitly checks for variable order. That would pass if we just picked the order from the first dataset as we do currently. 

So can we just take the order from the first dataset, and append all variables missing from the first dataset in an arbitrary order?

Reply:
Yes, the idea was to find the correct order, if at least one file contains all variables.

Sure, we can solely rely on the first dataset's order, if that's sufficient.

Reply:
> Sure, we can solely rely on the first dataset's order, if that's sufficient.

I think we should just take the simplest option, since there's no obvious better option. This is also nicely backwards compatible

Reply:
Now taking the order from dataset[0] and adding missing variables at the end.

Reply:
I wonder if we can simplify the order determination using something like
```python
import itertools

per_dataset_vars = [
    ["a", "d", "e"],
    ["b", "e", "c"],
    ["a", "f", "c"],
]
def unique_names(names):
    combined = itertools.chain.from_iterable(names)
    return list(dict.fromkeys(combined))

unique_names(per_dataset_vars)
# ['a', 'd', 'e', 'b', 'c', 'f']
```
where the names would be extracted using something like
```python
per_dataset_vars = [list(ds.variables) for ds in datasets]
```
I'm not sure what `data_names` or `max_vars_index` are for, though, so that might need some more thought

Reply:
Thanks @keewis, I'll have a look into this.